pkg/api/api.go

```diff
@@ -269,7 +269,7 @@ func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distrib
 		a.cfg.SkipLabelCountValidationHeader, limits, pushConfig.RetryConfig, d.PushWithMiddlewares, d.PushMetrics, a.logger,
 	), true, false, "POST")
 	a.RegisterRoute(OTLPPushEndpoint, distributor.OTLPHandler(
-		pushConfig.MaxOTLPRequestSize, d.RequestBufferPool, a.sourceIPs, limits, pushConfig.OTelResourceAttributePromotionConfig,
+		pushConfig.MaxOTLPRequestSize, d.RequestBufferPool, a.sourceIPs, limits, limits,
```

Comment:
@aknuds1 Arve is this correct? I understand that if there is no override for a user it will fallback to `pushConfig.OTelResourceAttributePromotionConfig`

Reply:
At least it looks weird to me to have to pass `limits` twice. Checking.

Reply:
This is because limits implements the interface that returns the `OTelResourceAttributePromotionConfig`